contracts/earn/strategies/Geist/StrategyGeist.sol

```diff
@@ -0,0 +1,278 @@
+        address token; // Deposit Token
+        address gToken; // Token that minted by lend
+    }
+
+    /* ----- State Variables ----- */
+
+    TokenAddresses public wantTokens;
+    TokenAddresses[] public rewards;
+
+    address[] public rewardToWNativeRoute;
+    address[][] public extraRewardToWNativeRoutes;
+    address[] public wNativeToWantRoute;
+
+    uint256 public lastHarvest;
+
+    /* ----- Constant Variables ----- */
+
+    address public constant wNative = address(0x21be370D5312f44cB42ce377BC9b8a0cEF1A4C83); // WFTM
+    address public constant reward = address(0xd8321AA83Fb0a4ECd6348D4577431310A6E0814d); // GEIST
+    address public constant uniRouter = address(0xF491e7B69E4244ad4002BC14e878a34207E38c29); // SpookySwapRouter
+
+    address public dataProvider = address(0xf3B0611e2E4D2cd6aB4bb3e01aDe211c3f42A8C3);
+    address public lendingPool = address(0x9FAD24f572045c7869117160A571B2e50b10d068);
+    address public multiFeeDistribution = address(0x49c93a95dbcc9A6A4D8f77E59c038ce5020e82f8);
+    address public incentivesController = address(0x297FddC5c33Ef988dd03bd13e162aE084ea1fE57);
+
+    /* ----- Events ----- */
+
+    event StratHarvest(address indexed harvester, uint256 wantHarvested, uint256 tvl);
+    event Deposit(uint256 tvl);
+    event Withdraw(uint256 tvl);
+
+    constructor(
+        address _vault,
+        address _accessManager,
+        address _want,
+        address[] memory _rewardToWNativeRoute,
+        address[][] memory _extraRewardToWNativeRoutes
+    ) public BaseStrategy(_vault, _accessManager) {
+        (address gToken, , ) = IDataProvider(dataProvider).getReserveTokensAddresses(_want);
+        wantTokens = TokenAddresses(_want, gToken);
+
+        rewardToWNativeRoute = _rewardToWNativeRoute;
+        extraRewardToWNativeRoutes = _extraRewardToWNativeRoutes;
+
+        for (uint256 i; i < extraRewardToWNativeRoutes.length; i++) {
+            address _token = extraRewardToWNativeRoutes[i][0];
+            (address _gToken, , ) = IDataProvider(dataProvider).getReserveTokensAddresses(_token);
+            rewards.push(TokenAddresses(_token, _gToken));
+        }
+
+        _giveAllowances();
+    }
+
+    /* ----- External Functions ----- */
+
+    function rewardToWNative() external view returns (address[] memory) {
+        return rewardToWNativeRoute;
+    }
+
+    /* ----- Public Functions ----- */
+
+    function harvest() public override whenNotPaused {
+        require(msg.sender == tx.origin || msg.sender == address(vault), 'StrategyGeist: EOA_or_vault');
+
+        uint256 gTokenBal = IERC20(wantTokens.gToken).balanceOf(address(this));
+        address[] memory tokens = new address[](1);
+        tokens[0] = wantTokens.gToken;
+        IIncentivesController(incentivesController).claim(address(this), tokens);
+        IMultiFeeDistribution(multiFeeDistribution).exit();
+
+        uint256 outputBal = IERC20(reward).balanceOf(address(this));
+        if (outputBal > 0) {
```

Comment:
Logic incorrect: remove this line, since there might be still other rewards to swap.